### ISSUE TITLE
Document errors should abort build

### ISSUE THREAD

**Issue by Y0hy0h** [dillonkearns/elm-pages#102]
When a document has an error, even when using `elm-pages build`, it is simply generated with the error text as page content.

I expect the build to fail, if not to at least give feedback on which pages failed to build, so that I can fix them.

## Reproduce
1. Clone the [elm-pages-starter](https://github.com/dillonkearns/elm-pages-starter) repo.
2. In `Main.elm`, replace the `markdownDocument` definition with:
    ```
    markdownDocument : { extension : String, metadata : Json.Decode.Decoder Metadata, body : String -> Result String (Element msg) }
    markdownDocument =
        { extension = "md"
        , metadata = Metadata.decoder
        , body =
            \markdownBody ->
                Err "I am displayed on the page."
        }
    ```
3. Run `npm run start` and open `localhost:3000`, or run `npm run build` and open `dist/index.html`. Both will show `I am displayed on the page.` as the body.

**Comment by dillonkearns** [comment]
Thank you so much for catching this, and for taking note of the steps to reproduce! I'm taking a look now.

In the future I'd like to get some automated end-to-end tests to check for these cases. I ran through a lot of cases manually before releasing, but I missed that one (for example, StaticHttp failures will stop the build). Markdown metadata failures are working as expected, too.

I'll keep you posted!

**Comment by dillonkearns** [comment]
Fixed in Elm package version 5.0.1 https://github.com/dillonkearns/elm-pages/blob/master/CHANGELOG-ELM.md#501---2020-05-13.

Let me know if it all works as expected now! Thanks again for catching that one 🙏

**Comment by Y0hy0h** [comment]
You're fixing things faster than I can break them. :D

I've noticed a source of confusion in the error message and opened #103.

This issue however is resolved, since parsing errors now behave like e.g. metadata errors.
